pkg/base/helm.go

```diff
@@ -271,3 +274,104 @@ func checkChartForVersion(file *upstreamtypes.UpstreamFile) (string, error) {
 	// if no determination is made, assume v2
 	return "v2", nil
 }
+
+// insert namespace if it's defined in the spec and not already present in the manifests
+func kustomizeHelmNamespace(baseFiles []BaseFile, renderOptions *RenderOptions) ([]BaseFile, error) {
+	if renderOptions.Namespace == "" {
+		return baseFiles, nil
+	}
+
+	chartsPath, err := ioutil.TempDir("", "charts")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create temp dir")
+	}
+	defer os.RemoveAll(chartsPath)
+
+	var updatedBaseFiles []BaseFile
+	var kustomizeResources []string
+	var kustomizePatches []kustomizetypes.PatchStrategicMerge
+	resources := map[string]BaseFile{}
+	foundGVKNamesMap := map[string]bool{}
+	for _, baseFile := range baseFiles {
+		// write temp files for manifests that need a namespace
+		gvk, manifest := GetGVKWithNameAndNs(baseFile.Content, renderOptions.Namespace)
+		if manifest.APIVersion == "" || manifest.Kind == "" || manifest.Metadata.Name == "" {
+			updatedBaseFiles = append(updatedBaseFiles, baseFile)
+			continue // ignore invalid resources
+		}
+
+		if manifest.Metadata.Namespace == "" {
+			name := filepath.Base(baseFile.Path)
+			tmpFile, err := ioutil.TempFile(chartsPath, name)
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to write temp file %v", tmpFile.Name())
+			}
```

Comment:
Since this is going to be a long running service, this should also defer close to prevent file handle leaks.  It's ok to call `Close` twice on a file.

Reply:
fixed